lang/cs/Org.Apache.REEF.IMRU.Examples/PipelinedBroadcastReduce/PipelinedBroadcastAndReduceFaultTolerant.cs

```diff
@@ -0,0 +1,158 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Org.Apache.REEF.Common.Tasks;
+using Org.Apache.REEF.IMRU.API;
+using Org.Apache.REEF.IMRU.OnREEF.Parameters;
+using Org.Apache.REEF.Tang.Annotations;
+using Org.Apache.REEF.Tang.Implementations.Configuration;
+using Org.Apache.REEF.Tang.Implementations.Tang;
+using Org.Apache.REEF.Tang.Interface;
+using Org.Apache.REEF.Tang.Util;
+using Org.Apache.REEF.Utilities.Diagnostics;
+using Org.Apache.REEF.Utilities.Logging;
+
+namespace Org.Apache.REEF.IMRU.Examples.PipelinedBroadcastReduce
+{
+    /// <summary>
+    /// IMRU program that performs broadcast and reduce with fault tolerant
+    /// </summary>
+    public class PipelinedBroadcastAndReduceFaultTolerant : PipelinedBroadcastAndReduce
+    {
+        private static readonly Logger Logger = Logger.GetLogger(typeof(PipelinedBroadcastAndReduceFaultTolerant));
+
+        [Inject]
+        protected PipelinedBroadcastAndReduceFaultTolerant(IIMRUClient imruClient) : base(imruClient)
+        {
+        }
+        
+        /// <summary>
+        /// Build a test mapper function configuration
+        /// </summary>
+        /// <param name="maxRetryInRecovery"></param>
```

Comment:
Can you please add a comment explaining what  this parameter means ? Is it a maximum number of retry attempts to re-execute mappers during recovery ? I am a bit confused by the "InRecovery" part of its name.